frontend/src/models/models.js

```diff
@@ -89,6 +89,37 @@ module.exports = app => app.component('models', {
     this.status = 'loaded';
   },
   methods: {
+    clickFilter(path) {
+      if (this.searchText) {
+        let searchObj;
+      
+        try {
+          // Try to parse this.searchText if it's a valid JSON string
+          searchObj = JSON.parse(this.searchText);
```

Comment:
This enforces `searchText` being strict JSON, which is definitely **not** something we want to do. For one thing, that makes filtering by ObjectId impossible in cases where Mongoose doesn't cast. For another thing, very inconvenient for the user to have to add double quotes around field names.